### PR TITLE
LPS-85972 Create and adapt unit tests for supporting LambdaFunctions of type ANY

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

**Comment by ruben-pulido** [comment]
ci:test:sf <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ruben-pulido** [comment]
ci:test:relevant <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ruben-pulido** [comment]
ci:test:sf <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ruben-pulido** [comment]
ci:test:relevant <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImpl.java [hunk 1]
```diff
@@ -74,6 +82,70 @@ public Filter visitBinaryExpressionOperation(
...
+	public EntityField visitLambdaVariableExpression(
+		LambdaVariableExpression lambdaVariableExpression) {
+
+		Map<String, EntityField> entityFieldsMap =
+			_entityModel.getEntityFieldsMap();
+
+		EntityField entityField = entityFieldsMap.get(
+			lambdaVariableExpression.getVariableName());
+
+		if (entityField == null) {
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImpl.java]: Throw ExpressionVisitorException

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/CollectionPropertyExpressionImpl.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	@Override
+	public PropertyExpression getPropertyExpression() {
+		return _propertyExpression;
+	}
+
+	@Override
+	public String toString() {
+		final StringBuilder sb = new StringBuilder(6);
+
+		sb.append("CollectionPropertyExpressionImpl{");
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/CollectionPropertyExpressionImpl.java]: Use the same toString pattern than in other classes

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/ComplexPropertyExpressionImpl.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	@Override
+	public PropertyExpression getPropertyExpression() {
+		return _propertyExpression;
+	}
+
+	@Override
+	public String toString() {
+		final StringBuilder sb = new StringBuilder(6);
+
+		sb.append("ComplexPropertyExpressionImpl{");
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/ComplexPropertyExpressionImpl.java]: Use same toString pattern than in the other classes

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/ExpressionVisitorImpl.java [hunk 4]
```diff
@@ -213,4 +286,54 @@ else if (binaryOperatorKind == BinaryOperatorKind.OR) {
 		return Optional.empty();
 	}
 
+	private PropertyExpression _getPropertyExpression(
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/ExpressionVisitorImpl.java]: Is it used?

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/LambdaFunctionExpressionImpl.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	@Override
+	public String getVariableName() {
+		return _variableName;
+	}
+
+	@Override
+	public String toString() {
+		final StringBuilder sb = new StringBuilder(8);
+
+		sb.append("LambdaFunctionExpressionImpl{");
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/LambdaFunctionExpressionImpl.java]: toString format

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/LambdaVariableExpressionImpl.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	@Override
+	public String getVariableName() {
+		return _variable;
+	}
+
+	@Override
+	public String toString() {
+		final StringBuilder sb = new StringBuilder(4);
+
+		sb.append("LambdaVariableExpressionImpl{");
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/LambdaVariableExpressionImpl.java]: Use same pattern

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/MemberExpressionImpl.java [hunk 7]
```diff
@@ -42,16 +51,31 @@ public MemberExpressionImpl(List<String> resourcePath) {
...
 	public List<String> getResourcePath() {
 		return _resourcePath;
 	}
 
 	@Override
 	public String toString() {
-		return _resourcePath.toString();
+		final StringBuilder sb = new StringBuilder(6);
+
+		sb.append("MemberExpressionImpl{");
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/MemberExpressionImpl.java]: Use same pattern

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/PrimitivePropertyExpressionImpl.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+	public String getName() {
+		return _name;
+	}
+
+	@Override
+	public String toString() {
+		final StringBuilder sb = new StringBuilder(4);
+
+		sb.append("PrimitivePropertyExpressionImpl{");
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/PrimitivePropertyExpressionImpl.java]: same pattern

**Comment by ruben-pulido** [comment]
Thanks for the feedback Cris! Changes done [here](https://github.com/cgoncas/liferay-portal/pull/307)

**Pull request closed without merging**